cfpb/consumerfinance.gov
# Use django-axes and Django password validators for custom password logic

This PR does two things:

1. Removes our custom login lockout functionality and adds the [django-axes](https://django-axes.readthedocs.io/en/latest/) library to handle that.
2. Moves our custom password requirements to [Django's `AUTH_PASSWORD_VALIDATORS ` functionality](https://docs.djangoproject.com/en/4.2/topics/auth/passwords/#module-django.contrib.auth.password_validation). 

Previously, our password validation happened within a subclass of the Django validator form, which required us to have a view that overrides Django/Wagtail's password change views. By using `AUTH_PASSWORD_VALIDATORS`, we can remove the custom views and form, and the URL overrides, and rely on Django to always check passwords through configured validators.

## How to test this PR

1. Use an incorrect password 5 times at http://localhost:8000/admin/login and confirm that you receive an account lockout.
2. Use the "Forgotten password?" link to reset your password. This will print the contents of the email it would send to your console. Attempt to reset your password to confirm that:

   - 12 characters are required
   - A capital letter is required
   - A lowercase letter is required
   - A number is required
   - A special character is required

## Checklist
